server/controllers/websocket/instrumented_mux.go

```diff
@@ -0,0 +1,35 @@
+package websocket
+
+import (
+	"net/http"
+	"sync/atomic"
+
+	"github.com/runatlantis/atlantis/server/logging"
+	"github.com/uber-go/tally"
+)
+
+type InstrumentedMultiplexor struct {
+	Multiplexor
+
+	numWsConnections int64
+	NumWsConnections tally.Gauge
+}
+
+func NewInstrumentedMultiplexor(multiplexor Multiplexor, statsScope tally.Scope, logger logging.SimpleLogging) Multiplexor {
+	return &InstrumentedMultiplexor{
+		Multiplexor:      multiplexor,
+		NumWsConnections: statsScope.SubScope("getprojectjobs").SubScope("websocket").Gauge("connections"),
```

Comment:
There's no point in doing 3 chained subscopes haha, the whole point of subscopes is so that other metrics can be linked on different levels to form a nice hierarchy of metrics to sift through. Nothing else can use the `getprojectjobs` subscope if you're doing that over here. Also, this mux doesn't really need to know about that scope at all since it shouldn't even know it's being invoked from that controller.

Let's pass in the `getprojectjobs` subscope at least.

Reply:
Also can you explain what the rational is in using `gauge` here? That's probably adding to your problem.  If stats are flushed every minute, then just use a counter and only increment?

Reply:
Yeah this makes sense! I've reorganized the jobs scope as follows: 
```
/jobs
   /log-streaming (Passed in to both the `InstrumentedMux` and `JobsController` )
      /websocket (Scope for multiplexor)
         /connections 
      /api (Scope for jobs controller)
         /ExecutionError
         /ExecutionTime
```

The reason we use a gauge here is because we want to increment and decrement the numOfWsConnections as connections are established and closed. Since the counters get reset every minute, I don't think using counters is feasible for our use case. We could keep track of numOfNewWsConnections using a counter but that wouldn't give us any info on whether the web socket connections were closed or not. 

In my opinion, we should use a gauge here. If a job takes longer than a minute to run, we would register opening and closing of web socket connections. However, if a jobs completes within a minute, we open and close the web socket connection immediately and the gauge does not pick up that change in value which I believe is fine since we have already closed the websocket connection.

Reply:
How is a counter not feasible? It's flushing metrics every minute so if you see the number staying steady or decreasing you know that connections are being closed.  

There's no point in having metrics around something if we're just going to drop them due to time constraints.  How would we be able to tell if we suddenly were spammed within a given minute with connection requests? We'd probably see indirect consequences from that but wouldn't be able to tie them to anything.

Reply:
Also `log-streaming` is not concise and descriptive enough here since it can be confused with system logs.  let's just remove that completely.

What i meant with my comment was just don't call the getprojectjobs subscope in this function itself, call it outside of it.

Reply:
Makes sense! I have changed it back to counter and moved the getprojectjobs scope outside.